### DISCUSSION TITLE
BW site award for the set which was revised

### DISCUSSION THREAD

**Discussion by julkasalsa** [RetroAchievements/RAWeb#692] [Ideas]
The site award could turn black and white if the game recieved revision. I think it's better solution instead of just make them vanish. Still it could be set in the profile, if somebody wouldn't like it.

**Comment by kdecks** [comment]
I don't think either should happen. As a player I like that when I've finished a set that I keep the trophy regardless off revision. Players would not like these changes.

**Comment by julkasalsa** [comment]
It might be optional. Some players would like to know if a set they completed got new achievements.

**Comment by Jamiras** [comment]
I'm not sure anyone would want their trophy to be demoted or disappear, but I could see wanting to revisit a modified set.

It would be better to address this with some sort of notification process. The simplest implementation would be to send a notification message to every user who has mastered the set. It would appear in their inbox and not affect their user page at all.

An alternate implementation would be to put a notification marker (!) over the icon to draw the user's attention. Upon hovering over it, they could see a message indicating that the set has been revised, and can dismiss the notification to make the marker disappear.

**Comment by timenoe** [comment]
I think a red border would be the best solution. The player will keep their badge, but the red border will bring to their attention that a set that have mastered has been revised, hence they no longer have mastered the set.

**Comment by redwizard42** [comment]
My thoughts: 
I think mastery should be full color, no border, and completion should be grayscale, no border.  With section for "former masteries/completions" where revised, but not re-mastered badges live (similar to how Events are split out).
Give people a profile-level setting to "[ ] automatically hide former masteries" or something so that can happen automagically if a user so chooses.

**Reply by redwizard42** [reply]
Just to be clear, the hiding would be for hiding on the public profile.

**Comment by Etron-RA** [comment]
I prefer the notification method Jamiras mentioned above.  If the mastery icon changed or dissappeared then I could foresee a lot of unfair hate directed towards the revision just because the player has to revisit the game.  Some games would require a full playthrough to re-master which would be a pain for a lengthy RPG since the player already put in the work once for the original mastery.  A dimissable notification would let the player know that a revision was made and the player doesn't feel forced to re-master game if they don't want to.

**Comment by Jamiras** [comment]
Another suggestion would be to put them in a separate section:
![image](https://user-images.githubusercontent.com/32680403/233850069-40210d29-bf1a-479a-878d-8c3032f11284.png)

Credit to DanielARP: https://discord.com/channels/310192285306454017/310195377993416714/1021536862306828380

**Reply by wescopeland** [reply]
I explored this solution in my implementation for https://github.com/RetroAchievements/RAWeb/pull/1503. However, I ultimately decided against going down this path. I received some feedback on the Discord that this could be undesirable, as users with many awards can sometimes be particular about their award sort order and this could perhaps be disruptive to that.

Similarly, my original implementation also put past awards in grayscale. However, after more discussion with community members, there was fear this could be perceived as a "cheapening" of the award that was indeed legitimately earned, oftentimes through hard work.
